src/Compilers/CSharp/Portable/Lowering/IteratorRewriter/IteratorRewriter.cs

```diff
@@ -137,34 +137,28 @@ private void GenerateEnumeratorImplementation()
             {
                 var disposeMethod = OpenMethodImplementation(
                     IDisposable_Dispose,
-                    debuggerHidden: true,
-                    generateDebugInfo: false,
                     hasMethodBodyDependency: true);
 
-                var moveNextMethod = OpenMethodImplementation(
-                    IEnumerator_MoveNext,
-                    methodName: WellKnownMemberNames.MoveNextMethodName,
-                    debuggerHidden: IsDebuggerHidden(this.method),
-                    hasMethodBodyDependency: true);
+                var moveNextMethod = OpenMoveNextMethodImplementation(IEnumerator_MoveNext);
 
                 GenerateMoveNextAndDispose(moveNextMethod, disposeMethod);
             }
 
             // Add T IEnumerator<T>.Current
             {
-                OpenPropertyImplementation(IEnumeratorOfElementType_get_Current, debuggerHidden: true, hasMethodBodyDependency: false);
```

Comment:
It isn't clear to me how the debuggerHidden behavior is preserved from the previous revision of the code in the absence of the scenario this PR is intended to change.